### PR TITLE
TDR-2538 - add FileMetadataFilters to get closure/descriptive metadata

### PR DESCRIPTION
**Pull request by vimleshtna**

### CODE REVIEW

**Review by TomJKing — COMMENTED** [review]
Looks good. One minor comment. Thanks

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala [hunk 1]
```diff
@@ -206,6 +208,20 @@ class FileService(
       rows <- fileRepository.getConsignmentParentFolder(consignmentId)
     } yield rows.map(_.fileid).headOption
   }
+
+  private def getPropertyNames(fileMetadataFilters: Option[FileMetadataFilters]): Future[Seq[String]] = {
```
[the remark this answers is not in the document]
Comment (vimleshtna) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala:212]: Yeah, good idea.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala [hunk 2]
```diff
@@ -275,6 +304,18 @@ object FileService {
     }
   }
 
+  implicit class CustomMetadataFieldsHelper(fields: Seq[FilepropertyRow]) {
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala:307]: Maybe rename the implicit class and functions to reflect this is handling `FilepropertyRow`s:

`implicit class FilePropertyRowsHelper(rows: Seq[FilepropertyRow]) ... etc ...`

**Review by TomJKing — COMMENTED** [review]
Sorry one more minor comment.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala (hunk 2 above)
Comment (vimleshtna) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala]: Yeah, sorry, I forgot to change the name.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**